File: prison-dashboard/src/data/prisons.ts
import type { JailStaff, PrisonRef } from "@/data/types"

/**
 * The jails and their staff. They match the backend's roster
 * (server/app/services/prisons.py); with a backend, sign-in is checked against the server.
 */
export const PRISONS: PrisonRef[] = [
  { id: "RNG-CJ", name: { en: "Rangpur Central Jail", bn: "রংপুর কেন্দ্রীয় কারাগার" } },
  { id: "NIL-DJ", name: { en: "Nilphamari District Jail", bn: "নীলফামারী জেলা কারাগার" } },
]

const prison = (id: string) => PRISONS.find((p) => p.id === id)!

export const JAIL_STAFF: JailStaff[] = [
  {
    id: "JS-03",
    name: { en: "Md. Golam Rabbani", bn: "মো. গোলাম রব্বানী" },
    designation: { en: "Deputy Jailer", bn: "ডেপুটি জেলার" },
    prison: prison("RNG-CJ"),
  },
  {
    id: "JS-08",
    name: { en: "Nasima Khatun", bn: "নাসিমা খাতুন" },
    designation: { en: "Legal Aid Desk Officer", bn: "লিগ্যাল এইড ডেস্ক কর্মকর্তা" },
    prison: prison("RNG-CJ"),
  },
  {
    id: "JS-12",
    name: { en: "Md. Shafiqul Alam", bn: "মো. শফিকুল আলম" },
    designation: { en: "Jailer", bn: "জেলার" },
    prison: prison("NIL-DJ"),
  },
]

export function findStaff(id: string): JailStaff | undefined {
  const key = id.trim().toUpperCase()
  return JAIL_STAFF.find((s) => s.id === key)
}

/** The two demo accounts, both at Rangpur Central Jail. */
export const DEMO_STAFF = { desk: "JS-08", deputy: "JS-03" } as const
export const DEMO_PASSWORD = "demo1234"
export const MIN_PASSWORD_LENGTH = 4

File: server/app/services/prisons.py
"""The jails whose staff use prison-dashboard, and those staff.

The roster is kept here and matches prison-dashboard's own list
(``src/data/prisons.ts``). Adding a jail or a member of staff is not available
from the API yet.

Staff sign in on prison-dashboard: requests name them in
``X-Prison-Staff-Id``, which must be on this roster.
"""

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class Prison:
    id: str
    name: str
    name_bn: str

    def ref(self) -> dict[str, Any]:
        return {"id": self.id, "name": self.name, "nameBn": self.name_bn}


@dataclass(frozen=True)
class PrisonStaff:
    id: str
    name: str
    name_bn: str
    designation: str
    designation_bn: str
    prison_id: str

    @property
    def prison(self) -> Prison:
        return PRISONS_BY_ID[self.prison_id]

    @property
    def actor(self) -> str:
        """How they appear in the audit ledger."""
        return f"prison:{self.id}"

    def view(self) -> dict[str, Any]:
        return {
            "id": self.id,
            "name": self.name,
            "nameBn": self.name_bn,
            "designation": self.designation,
            "designationBn": self.designation_bn,
            "prison": self.prison.ref(),
        }


PRISONS: tuple[Prison, ...] = (
    Prison("RNG-CJ", "Rangpur Central Jail", "রংপুর কেন্দ্রীয় কারাগার"),
    Prison("NIL-DJ", "Nilphamari District Jail", "নীলফামারী জেলা কারাগার"),
)

PRISONS_BY_ID = {prison.id: prison for prison in PRISONS}

PRISON_STAFF: tuple[PrisonStaff, ...] = (
    PrisonStaff(
        "JS-03", "Md. Golam Rabbani", "মো. গোলাম রব্বানী", "Deputy Jailer", "ডেপুটি জেলার", "RNG-CJ"
    ),
    PrisonStaff(
        "JS-08",
        "Nasima Khatun",
        "নাসিমা খাতুন",
        "Legal Aid Desk Officer",
        "লিগ্যাল এইড ডেস্ক কর্মকর্তা",
        "RNG-CJ",
    ),
    PrisonStaff("JS-12", "Md. Shafiqul Alam", "মো. শফিকুল আলম", "Jailer", "জেলার", "NIL-DJ"),
)

_STAFF_BY_ID = {staff.id: staff for staff in PRISON_STAFF}


def get_prison(prison_id: str) -> Prison | None:
    return PRISONS_BY_ID.get(prison_id.strip().upper())


def get_prison_staff(staff_id: str) -> PrisonStaff | None:
    return _STAFF_BY_ID.get(staff_id.strip().upper())
